File: backend/internal/domain/selection.go
package domain

// Selection holds the currently active project/environment/cloud/location
// chosen in the UI. Values are stored by name; "any" means no specific choice.
type Selection struct {
	Project     string `json:"project"`
	Environment string `json:"environment"`
	Cloud       string `json:"cloud"`
	Location    string `json:"location"`
}

File: frontend/src/features/projects/api/selection.ts
import { api } from '@/lib/api-client'

export interface Selection {
	project: string
	environment: string
	cloud: string
	location: string
}

export function getSelection(): Promise<Selection> {
	return api.get<Selection>('/selection')
}

export function setSelection(selection: Selection): Promise<Selection> {
	return api.put<Selection>('/selection', selection)
}
